### PR TITLE
Build full test suite into a single bundle

### PR DESCRIPTION
**Pull request by jridgewell**
For larger projects, having each test file compile its own copy of the entire codebase leads to gigantic builds (AMP reached 20+ GB before my computer killed the process). This is because we treated each file as its own distinct `entryPoint`.

Instead, this creates a single bundle file that imports each test file, and has esbuild build that file. This leads to a single copy of the code base, and considerably smaller builds.

### CODE REVIEW

**Comment by jridgewell** [comment]
This should wait for #12 before we merge, so I can ensure the test suite passes on all OSs.

**Comment by jridgewell** [comment]
I had to include #13 in this PR to get the tests to consistently pass. You can see a diff by going to https://github.com/jridgewell/karma-esbuild/compare/karma-jsdom-launcher...single-bundle.

File: src/index.ts [hunk 1]
```diff
@@ -30,22 +30,43 @@ interface KarmaLogger {
...
-	// Inject sourcemap middleware
+	// Inject middleware to handle the bundled file and map.
 	if (!config.middleware) {
 		config.middleware = [];
 	}
 	config.middleware.push("esbuild");
 
+	// Create an empty file for Karma to track. Karma requires a real file in
+	// order for it to be injected into the page, even though the middleware
+	// will be responsible for serving it.
```
Comment (marvinhagemeister) [inline src/index.ts:53]: Love these comments 👍

File: src/index.ts [hunk 2]
```diff
@@ -96,22 +111,30 @@ function createPreprocessor(
...
+	async function build(contents: string, file: string) {
 		const userConfig = { ...config.esbuild };
 
 		const result = await service!.build({
 			target: "es2015",
 			...userConfig,
 			bundle: true,
 			write: false,
-			entryPoints: [file],
+			stdin: {
```
Comment (marvinhagemeister) [inline src/index.ts:133]: TIL about `stdin`. Didn't know that esbuild supports that 👍

**Review by marvinhagemeister — APPROVED** [review]

**Pull request merged**